connector/src/main/java/org/jboss/as/connector/subsystems/datasources/DataSourceModelNodeUtil.java

```diff
@@ -288,8 +287,7 @@ static ModifiableXaDataSource xaFrom(final OperationContext operationContext, fi
         final DsSecurity security = new DsSecurityImpl(username, password,
                 elytronEnabled? authenticationContext: securityDomain, elytronEnabled, credentialSourceSupplier, reauthPlugin);
 
-        final Boolean sharePreparedStatements = dataSourceNode.hasDefined(SHARE_PREPARED_STATEMENTS.getName()) ? dataSourceNode.get(
-                SHARE_PREPARED_STATEMENTS.getName()).asBoolean() : Defaults.SHARE_PREPARED_STATEMENTS;
+        final Boolean sharePreparedStatements = ModelNodeUtil.getBooleanIfSetOrGetDefault(operationContext, dataSourceNode, SHARE_PREPARED_STATEMENTS);
```

Comment:
This also should never return null.  I assume this line is the actual bug fix; replacing the custom code with code that calls SHARE_PREPARED_STATEMENTS.resolveModelAttribute.

Ideally most/all of these ModelNodeUtil.getXXXIfSetOrGetDefault call should be replaced by AttributeDefinition.resolveModelAttribute(context, modelnode).asXXX() or .asXXXOrNull() if the AD isn't required and has no default value.  That's out of scope for this though. :)

Reply:
OK, I will use `resolveModelAttribute().asBoolean()` at least for both SHARE_PREPARED_STATEMENTS readings.